### PR TITLE
sort logical sensors according to their dependency

### PR DESCRIPTION
**Pull request by OlivierWienholtz**
Sort logical sensors according to their dependency in Starter 
Before explicit loop, sensor_time & logical sensors are called :  Some may be active at that time
Logic sensor  code has been refactored.

#### Checklist

- [X] The title of the PR is reviewed
- [X] There is no text before the checklist section
- [X] The following two sections are filled (or deleted)
- [X] This PR has been previewed

### CODE REVIEW

File: engine/source/tools/sensor/sensor_and.F [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+C   C o m m o n   B l o c k s
+C-----------------------------------------------
+#include      "com04_c.inc"
+#include      "com08_c.inc"
+#include      "units_c.inc"
+#include      "task_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER SENS
```
Comment (sebastienVilleneuve) [inline engine/source/tools/sensor/sensor_and.F:61]: INTENT are missing here

File: engine/source/tools/sensor/sensor_not.F [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+C   C o m m o n   B l o c k s
+C-----------------------------------------------
+#include      "com04_c.inc"
+#include      "com08_c.inc"
+#include      "units_c.inc"
+#include      "task_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER SENS
```
Comment (sebastienVilleneuve) [inline engine/source/tools/sensor/sensor_not.F:61]: INTENT are missing here

File: engine/source/tools/sensor/sensor_or.F [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+C   C o m m o n   B l o c k s
+C-----------------------------------------------
+#include      "com04_c.inc"
+#include      "com08_c.inc"
+#include      "units_c.inc"
+#include      "task_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER SENS
```
Comment (sebastienVilleneuve) [inline engine/source/tools/sensor/sensor_or.F:61]: INTENT are missing here

File: engine/source/tools/sensor/sensor_sens.F [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+C   C o m m o n   B l o c k s
+C-----------------------------------------------
+#include      "com04_c.inc"
+#include      "com08_c.inc"
+#include      "units_c.inc"
+#include      "task_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER SENS
```
Comment (sebastienVilleneuve) [inline engine/source/tools/sensor/sensor_sens.F:79]: INTENT are missing here

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
Hi @OlivierWienholtz 

some little correction to perform

**Review by sebastienVilleneuve — APPROVED** [review]

File: starter/source/tools/sensor/sort_logical_sensors.F [hunk 5]
```diff
@@ -0,0 +1,167 @@
+Copyright>        OpenRadioss
+Copyright>        Copyright (C) 1986-2022 Altair Engineering Inc.
```
[the remark this answers is not in the document]
Comment (OlivierWienholtz) [reply starter/source/tools/sensor/sort_logical_sensors.F]: Done, Thank You,

File: engine/source/tools/sensor/sensor_sens.F [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR), TARGET :: SENSOR_TAB
```
[the remark this answers is not in the document]
Comment (OlivierWienholtz) [reply engine/source/tools/sensor/sensor_sens.F]: Done

**Review by sebastienVilleneuve — DISMISSED** [review]

**Comment by OlivierWienholtz** [comment]
Checked the results..
2 QA test suite models updated. Prior : 1 cycle difference between /SENSOR/SENS activation and its underlying sensor
Now it is the same cycle.

**Review by sebastienVilleneuve — APPROVED** [review]

**Pull request merged**